e2eAIOK/DeNas/nlp/bert_trainer.py

```diff
@@ -60,14 +73,20 @@ def _post_process(self):
     def train_one_epoch(self, epoch):
         # set random seed
         # random.seed(epoch)
-        if self.train_dataloader.sampler is not None and hasattr(self.train_dataloader.sampler, "set_epoch"):
-            self.train_dataloader.sampler.set_epoch(epoch)
+        
+        #if self.train_dataloader.sampler is not None and hasattr(self.train_dataloader.sampler, "set_epoch"):
+        #    self.train_dataloader.sampler.set_epoch(epoch)
+        if hasattr(self.train_dataloader.dataset, "set_epoch"):
+            self.train_dataloader.dataset.set_epoch(epoch)
 
         for step, batch in enumerate(tqdm(self.train_dataloader, desc="Iteration", ascii=True)):
             self.model.train()
             inputs, targets = batch
             outputs = self.model(inputs)
-            loss = self.criterion(outputs, targets)
+            if self.cfg.is_tl:
+                loss = self.model.loss(outputs, targets)
```

Comment:
any chance we can merge this two option? looks to me that using 'model.loss' can also means using 'criterion' ? so we don't need to add a switch here?

Reply:
Got it, I will refine this issue. Thx.